html_text/html_text.py

```diff
@@ -47,45 +49,71 @@ def parse_html(html):
 _has_punct_before = re.compile(r'\($').search
 
 
-def selector_to_text(sel, guess_punct_space=True):
```

Comment:
This function was useful - the main use case is to extract text from a part of a web page, finding this part using Scrapy or parsel.

Reply:
You can get a parsed tree for a selector using `sel.root`

Reply:
Ok so you want to create a selector in `extract_text(html)` and then apply `traverse_text_fragments` on `sel.root`, correct?

Reply:
if `selector_to_text` is supported, `cleaned_selector` is also nice to have

Reply:
@Kebniss no, extract_text doesn't need to use Selector, it is an additional overhead. The idea is to be backwards compatible and provide the same feature for Selector; internally it can work the other way around - likely selector_to_text should pass sel.root to html_to_text.

Reply:
👍

Reply:
It is not possible to just call `html_to_text` from inside `selector_to_text` because for each node `html_to_text`  extracts both the node's text and its tail, this can return extra bits of text when we are using selectors to extract only the text within specific nodes. I thought of adding an argument from `selector_to_text` to `html_to_text` with the address to `sel.root`, this would allow the algorithm to understand when it is back to the original node and not to extract tail in this case. However this feels very much as an hack and I think there will be cases where we would not want to extract tail text from subnodes of root and this would do nothing to avoid it.

I am not sure how to go around this: if we want to keep backward compatibility with selectors we have to use the old extraction method but in this algorithm it's not possible to know node's tag and implement `guess_page_layout`. Would it be acceptable to provide two independent pathways one using selectors and one that allows to use `guess_page_layout` on the whole html?

Reply:
Can't we have an `tail_text` argument in html_to_text? When it is False, tail extraction is skipped, but only on top level (i.e. text is still extracted from children tails). It can be False by default - I don't see why would anyone want to extract text from the tail of the root element.

Reply:
Alternatively, we can just not extract text from element tail by default, on the top level (i.e. children should have tail text processed as usual). 

In a common case (root `<html>` element) there shouldn't be any text in the tail. And when user passes another element explicitly, extracting text from element tail is likely undesirable - it is the same issues as with Selectors.

Reply:
you are right, tail text is outside selected nodes and as such it should not be extracted. Not extracting it by default seems reasonable. I will add the root object as argument to check when the recursion call is processing it

Reply:
Why do you need a root object, can't is just be a boolean flag `process_tail`?

Reply:
yy, root node is unnecessary. I added a depth argument so that we know when the recursion is back to root and does not extract tail there